ocaml/xapi/xapi_vm_migrate.ml

```diff
@@ -1002,36 +1002,46 @@ let assert_can_migrate  ~__context ~vm ~dest ~live ~vdi_map ~vif_map ~options =
       `cross_pool
   in
 
-  (* Check VDIs are not migrating to or from an SR which doesn't have required_sr_operations *)
+  (* operations required for migration *)
   let required_sr_operations = [Smint.Vdi_mirror; Smint.Vdi_snapshot] in
-  assert_sr_support_operations ~__context ~vdi_map ~remote ~ops:required_sr_operations;
```

Comment:
This is the reason indeed. The aim of this change is to have the sr check after the target host version check, which exists in each type (of course, you may argue this can be factored out too, which I can do if necessary).